### ISSUE TITLE
Couldn’t access to my notifications channels 

### ISSUE THREAD

**Issue by Daly123456** [grafana/grafana#28431]
why after upgrading to Grafana 7.2.0 i got a lot of bug likethis one
1-i couldn’t access my notification channels
2-some dashboards couldn’t be loaded and i need each time to re-import it and overwrite it to resolve that problem
I’am stuuuck here some help please !
how can i solve that please
![aas](https://user-images.githubusercontent.com/46686877/96709479-a50a4a00-1392-11eb-9185-f3b61452d15c.PNG)

**What happened**:

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Grafana version:
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

**Comment by torkelo** [comment]
Can you show your browser console? and the result of the network api call when you open this view

**Comment by Daly123456** [comment]
![image](https://user-images.githubusercontent.com/46686877/96849926-21ae2e80-144e-11eb-9f71-ae871860cda6.png)

**Comment by Daly123456** [comment]
![image](https://user-images.githubusercontent.com/46686877/96850026-41455700-144e-11eb-8bb6-437a736ad21e.png)

**Comment by Daly123456** [comment]
![image](https://user-images.githubusercontent.com/46686877/96850149-65a13380-144e-11eb-86e5-8fc1b021dc95.png)

**Comment by Daly123456** [comment]
![image](https://user-images.githubusercontent.com/46686877/96850292-8cf80080-144e-11eb-9c82-d0f16901ce2f.png)
i got this both errors in two differents organisations

**Comment by inkdude** [comment]
I have same problem
I copy json dashboards (from Grafana v6 to Grafana v7.2) used provision

In logs i see

```
t=2020-10-23T08:28:11+0000 lvl=dbug msg="Start walking disk" logger=provisioning.dashboard type=file name=default path=/var/lib/grafana/dashboards
t=2020-10-23T08:28:11+0000 lvl=dbug msg="Failed to translate Notification Id to Uid" logger=alerting.testnotification orgID=1 Id=4
t=2020-10-23T08:28:11+0000 lvl=eror msg="Unable to translate notification id to uid" logger=alerting.testnotification error="Alert notification [ Id: 4, OrgId: 1 ] not found" dashboardId=2 alertId=1 panelId=14 notificationId=4
t=2020-10-23T08:28:11+0000 lvl=dbug msg="Failed to translate Notification Id to Uid" logger=alerting.testnotification orgID=1 Id=4
t=2020-10-23T08:28:11+0000 lvl=eror msg="Unable to translate notification id to uid" logger=alerting.testnotification error="Alert notification [ Id: 4, OrgId: 1 ] not found" dashboardId=2 alertId=2 panelId=72 notificationId=4
t=2020-10-23T08:28:11+0000 lvl=dbug msg="Failed to translate Notification Id to Uid" logger=alerting.testnotification orgID=1 Id=4
t=2020-10-23T08:28:11+0000 lvl=eror msg="Unable to translate notification id to uid" logger=alerting.testnotification error="Alert notification [ Id: 4, OrgId: 1 ] not found" dashboardId=2 alertId=4 panelId=76 notificationId=4
t=2020-10-23T08:28:11+0000 lvl=dbug msg="Failed to translate Notification Id to Uid" logger=alerting.testnotification orgID=1 Id=4
t=2020-10-23T08:28:11+0000 lvl=eror msg="Unable to translate notification id to uid" logger=alerting.testnotification error="Alert notification [ Id: 4, OrgId: 1 ] not found" dashboardId=2 alertId=5 panelId=74 notificationId=4
t=2020-10-23T08:28:11+0000 lvl=dbug msg="Failed to translate Notification Id to Uid" logger=alerting.testnotification orgID=1 Id=4
t=2020-10-23T08:28:11+0000 lvl=eror msg="Unable to translate notification id to uid" logger=alerting.testnotification error="Alert notification [ Id: 4, OrgId: 1 ] not found" dashboardId=2 alertId=7 panelId=16 notificationId=4
t=2020-10-23T08:28:11+0000 lvl=dbug msg="Failed to translate Notification Id to Uid" logger=alerting.testnotification orgID=1 Id=4
t=2020-10-23T08:28:11+0000 lvl=eror msg="Unable to translate notification id to uid" logger=alerting.testnotification error="Alert notification [ Id: 4, OrgId: 1 ] not found" dashboardId=2 alertId=8 panelId=4 notificationId=4
t=2020-10-23T08:28:11+0000 lvl=dbug msg="Failed to translate Notification Id to Uid" logger=alerting.testnotification orgID=1 Id=4
t=2020-10-23T08:28:11+0000 lvl=eror msg="Unable to translate notification id to uid" logger=alerting.testnotification error="Alert notification [ Id: 4, OrgId: 1 ] not found" dashboardId=2 alertId=10 panelId=10 notificationId=4
t=2020-10-23T08:28:11+0000 lvl=dbug msg="Failed to translate Notification Id to Uid" logger=alerting.testnotification orgID=1 Id=4
t=2020-10-23T08:28:11+0000 lvl=eror msg="Unable to translate notification id to uid" logger=alerting.testnotification error="Alert notification [ Id: 4, OrgId: 1 ] not found" dashboardId=9 alertId=16 panelId=16 notificationId=4
t=2020-10-23T08:28:11+0000 lvl=dbug msg="Failed to translate Notification Id to Uid" logger=alerting.testnotification orgID=1 Id=4
t=2020-10-23T08:28:11+0000 lvl=eror msg="Unable to translate notification id to uid" logger=alerting.testnotification error="Alert notification [ Id: 4, OrgId: 1 ] not found" dashboardId=9 alertId=19 panelId=14 notificationId=4
t=2020-10-23T08:28:11+0000 lvl=dbug msg="Failed to translate Notification Id to Uid" logger=alerting.testnotification orgID=1 Id=4
t=2020-10-23T08:28:11+0000 lvl=eror msg="Unable to translate notification id to uid" logger=alerting.testnotification error="Alert notification [ Id: 4, OrgId: 1 ] not found" dashboardId=9 alertId=20 panelId=8 notificationId=4
t=2020-10-23T08:28:11+0000 lvl=dbug msg="Failed to translate Notification Id to Uid" logger=alerting.testnotification orgID=1 Id=4
t=2020-10-23T08:28:11+0000 lvl=eror msg="Unable to translate notification id to uid" logger=alerting.testnotification error="Alert notification [ Id: 4, OrgId: 1 ] not found" dashboardId=9 alertId=21 panelId=10 notificationId=4
t=2020-10-23T08:28:11+0000 lvl=dbug msg="Failed to translate Notification Id to Uid" logger=alerting.testnotification orgID=1 Id=4
t=2020-10-23T08:28:11+0000 lvl=eror msg="Unable to translate notification id to uid" logger=alerting.testnotification error="Alert notification [ Id: 4, OrgId: 1 ] not found" dashboardId=9 alertId=22 panelId=4 notificationId=4
t=2020-10-23T08:28:11+0000 lvl=dbug msg="Failed to translate Notification Id to Uid" logger=alerting.testnotification orgID=1 Id=4
t=2020-10-23T08:28:11+0000 lvl=eror msg="Unable to translate notification id to uid" logger=alerting.testnotification error="Alert notification [ Id: 4, OrgId: 1 ] not found" dashboardId=9 alertId=23 panelId=2 notificationId=4
```

I user ansible cloudalchemy grafana role

```
grafana_alert_notifications:
  notifiers:
    - name: Slack
      type: slack
      uid: 16xNZWeZz
      is_default: true
      send_reminder: true
      frequency: 1h
      disable_resolve_message: false
      settings:
        icon_url: "https://avatars3.githubusercontent.com/u/3380462"
        recipient: "#it_notify_office"
        uploadImage: true
      secure_settings:
        url: "XXX"
```

Here REST API query for alerts in old grafana (v6)

```
[
  {
    "id": 4,
    "uid": "16xNZWeZz",
    "name": "Slack",
    "type": "slack",
    "isDefault": false,
    "sendReminder": true,
    "disableResolveMessage": false,
    "frequency": "1h",
    "created": "2020-04-28T15:08:22+03:00",
    "updated": "2020-06-10T16:34:45+03:00",
    "settings": {
      "autoResolve": true,
      "httpMethod": "POST",
      "icon_url": "https://avatars3.githubusercontent.com/u/3380462",
      "mention": "",
      "recipient": "#XXX",
      "token": "",
      "uploadImage": true,
      "url": "XXX"
    }
  },
  {
    "id": 5,
    "uid": "_Mq-1cmMk",
    "name": "web",
    "type": "slack",
    "isDefault": false,
    "sendReminder": false,
    "disableResolveMessage": false,
    "frequency": "",
    "created": "2020-06-08T13:59:29+03:00",
    "updated": "2020-06-11T12:21:27+03:00",
    "settings": {
      "autoResolve": true,
      "httpMethod": "POST",
      "mention": "@channel",
      "recipient": "#XXX",
      "token": "XXX",
      "uploadImage": false,
      "url": "XXX"
    }
  },
  {
    "id": 3,
    "uid": "paM9HrXWk",
    "name": "web",
    "type": "telegram",
    "isDefault": false,
    "sendReminder": false,
    "disableResolveMessage": false,
    "frequency": "",
    "created": "2020-03-19T16:44:39+03:00",
    "updated": "2020-03-19T16:44:39+03:00",
    "settings": {
      "autoResolve": true,
      "bottoken": "XXX",
      "chatid": "XXX",
      "httpMethod": "POST",
      "uploadImage": true
    }
  }
]
```

What does it mean - id, uid? In new grafana something about it is changed? Can you explain me?

I can't open notification channel from web interface

**Comment by peterholmberg** [comment]
> 1-i couldn’t access my notification channels

This has been fixed in 7.3 https://github.com/grafana/grafana/pull/28115

**Comment by marefr** [comment]
Regarding "Unable to translate notification id to uid" it means that you've specified a notification channel id in your dashboard that doesn't exist. When using provisioning of alert notification channels and dashboards it's important that you refer to notification channel uid in dashboards because the id (database incremental counter) is non-deterministic where's the uid is deterministic/stable.

@papagian anything to add here? Otherwise I think we can close it

**Comment by papagian** [comment]
The original issue (unable to load the notification channel) is duplicate of #28051 (as @peterholmberg has pointed out).

Regarding the errors in the logs, on top of what @marefr's comment, these errors are not related to the original issue above but imply that inside the dashboard JSON the notification channels (under `notifications` section) are referred using the notification id instead of uid. Therefore, grafana tries to convert ids to uids but is unable to find the specific notification channel.
grafana 7.3 contains some improvements for handling this:
- the frequency of these errors is reduced by logging a warning only when an alert is saved or tested and not in each alert evaluation #28190.
- when the user navigates to the alert tab (under a specific panel) and such an outdated notification channel is detected then a confirmation modal prompts for deleting it #28190.

In conclusion, the experience is expected to be improved in 7.3 and given all the issues are addressed in 7.3 I think that we can close it.
